### PR TITLE
chore: remove support for netlify.json/.yaml/.yml

### PR DESCRIPTION
**Pull request by 4xposed**
Addresses: https://github.com/netlify/netlify-commons/issues/347

When looking into https://github.com/netlify/buildbot/issues/2141 we saw that files in `yaml`, `yml` and `json` are still supported in `netlify-commons/ntoml` package.

But in `build` we only make use of `netlify.toml` https://github.com/netlify/build/blob/60523200de880d86716acb6079e2699e32cd90f9/packages/config/src/path.js#L52

This PR removes all logic related to support the different .yaml/.yml/.json files

### CODE REVIEW

File: ntoml/netlify_toml.go [hunk 1]
```diff
@@ -148,12 +115,6 @@ func LoadFrom(paths ...string) (*NetlifyToml, error) {
 			switch extension {
 			case ".toml":
```
Comment (mraerino) [inline ntoml/netlify_toml.go:116]: could you maybe simplify this code since we don't really care about different extensions anymore?

**Review by mraerino — COMMENTED** [review]
mostly good, one thing

File: ntoml/netlify_toml.go (hunk 1 above)
Comment (4xposed) [reply ntoml/netlify_toml.go]: done 👍

**Review by mraerino — APPROVED** [review]
LGTM

**Pull request merged**